lib/GoalCohort.js

```diff
@@ -76,6 +76,7 @@ const GoalCohort = new Schema(
       required: "{PATH} is required!",
       index: true
     },
+    name: { type: String },
```

Comment:
If we're going to have `name` then I think we want to forbid the (the canonical version of each) name from being used more than once in association with each goal...

To do this starts with a tests like:

```
[
  {name: 'chiefs', duplicate: 'chiefs'},
  {name: 'Chiefs', duplicate: 'cHiefs'},
  {name: 'the chiefs', duplicate: 'thechiefs'}
].forEach(ex, ()=> {
  it('throws an error when user attempts to create a second cohort with a given name for the same goal', async () => {
  }
}
```

...we also will need a test that ensures that you can create two distinct GoalCohort instances for the same goal and each having NO name (to ensure we're not considering `null` a unique name that can only be used once per goal.

To make the all of above work,  you would probably need two things:

- a `nameCanonical` field that stores the lower/canonicalized version of the name
- a new index constraint like

```
GoalCohort.index({ 
  goal: 1, 
  nameCanonical: 1 
}, { 
  unique: true,
  sparse: true // because we allow null/undefined name
});
```